processor/lsmintervalprocessor/internal/merger/model.go

```diff
@@ -0,0 +1,389 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package merger // import "github.com/elastic/opentelemetry-collector-components/processor/lsmintervalprocessor/internal/merger"
+
+import (
+	"encoding/binary"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/elastic/opentelemetry-collector-components/processor/lsmintervalprocessor/internal/identity"
+)
+
+// TODO: Think about multitenancy, should be part of the key
```

Comment:
> I think we briefly discussed this but I can't recall the details of our discussion: an alternative may be to create a separate directory/PebbleDB instance per tenant

Yeah, but I don't think we concluded anything. This is one way, however, we would need to identify the tenant first. Also, I think using key taints is better than separate directory in pebble since the keys would partition the data and there is almost negligible chances of any data mixing between tenants due to how data is kept at rest.

> We can keep memory usage in check by limiting concurrency of merges.

Currently, we have this config as `1`, so no concurrent merges.